pkg/shell/manifests.ts

```diff
@@ -19,60 +19,129 @@
-    path?: string;
-    label?: string;
-    order?: number;
-    docs?: ManifestDocs[];
-    keywords?: ManifestKeyword[];
+    path: string | undefined;
+    label: string | undefined;
+    order: number | undefined;
+    docs: ManifestDocs[] | undefined;
+    keywords: ManifestKeyword[] | undefined;
+}
+
+function import_ManifestEntry(val: JsonValue): ManifestEntry {
+    const obj = import_json_object(val);
+    return {
+        path: import_optional(obj, "path", import_string),
+        label: import_optional(obj, "label", import_string),
+        order: import_optional(obj, "order", import_number),
+        docs: import_optional(obj, "docs", v => import_array(v, import_ManifestDocs)),
+        keywords: import_optional(obj, "keywords", v => import_array(v, import_ManifestKeyword)),
+    };
 }
 
 export interface ManifestSection {
     [name: string]: ManifestEntry;
 }
 
+function import_ManifestSection(val: JsonValue): ManifestSection {
+    return import_record(val, import_ManifestEntry);
+}
+
 export interface ManifestParentSection {
-    component?: string;
-    docs?: ManifestDocs[];
+    component: string | undefined;
+    docs: ManifestDocs[] | undefined;
+}
+
+function import_ManifestParentSection(val: JsonValue): ManifestParentSection {
+    const obj = import_json_object(val);
+    return {
+        component: import_optional(obj, "component", import_string),
+        docs: import_optional(obj, "docs", v => import_array(v, import_ManifestDocs))
+    };
 }
 
 export interface Manifest {
-    dashboard?: ManifestSection;
-    menu?: ManifestSection;
-    tools?: ManifestSection;
+    dashboard: ManifestSection | undefined;
+    menu: ManifestSection | undefined;
+    tools: ManifestSection | undefined;
+
+    preload: string[] | undefined;
+    parent: ManifestParentSection | undefined;
+    ".checksum": string | undefined;
+}
 
-    preload?: string[];
-    parent?: ManifestParentSection;
-    ".checksum"?: string;
+function import_Manifest(val: JsonValue): Manifest {
+    const obj = import_json_object(val);
+    return {
+        dashboard: import_optional(obj, "dashboard", import_ManifestSection),
+        menu: import_optional(obj, "menu", import_ManifestSection),
+        tools: import_optional(obj, "tools", import_ManifestSection),
+        preload: import_optional(obj, "preload", v => import_array(v, import_string)),
+        parent: import_optional(obj, "parent", import_ManifestParentSection),
+        ".checksum": import_optional(obj, ".checksum", import_string),
+    };
 }
 
 export interface Manifests {
     [pkg: string]: Manifest;
 }
 
 export function import_Manifests(val: JsonValue): Manifests {
-    // TODO - validate against schema
-    return val as unknown as Manifests;
+    const obj = import_json_object(val);
+    const res: Manifests = { };
+    for (const pkg in obj) {
+        const m = validate("." + pkg, obj[pkg], import_Manifest, null);
+        if (m)
+            res[pkg] = m;
```

Comment:
Ah, so this takes care of (entirely) ignoring broken manifests but keeping the good ones. For a broken manifest, `validate()` would return the `null` default value that you gave it, right? And the importer already does `console.error()` to notify the user what's wrong. Did I get that right?

Can you please add an integration test (`check-shell-menu` feels quite appropriate, in testBasic()?) that adds an invalid manifest and ensures that (1) it's not shown in the menu, and (2) other menu items are?